pkgs/by-name/tr/tracy/default.nix

```diff
@@ -0,0 +1,131 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  cmake,
+  ninja,
+
+  capstone,
+  darwin,
+  dbus,
+  freetype,
+  glfw,
+  pkg-config,
+  tbb,
+
+  withWayland ? stdenv.isLinux,
+  libglvnd,
+  libxkbcommon,
+  wayland-protocols,
+  wayland,
+}:
+
+stdenv.mkDerivation rec {
+  pname = if withWayland then "tracy-wayland" else "tracy-glfw";
+  version = "0.11.0";
+
+  src = fetchFromGitHub {
+    owner = "wolfpld";
+    repo = "tracy";
+    rev = "v${version}";
+    sha256 = "sha256-wGKnmCpmG70Xx/+RfZhXTLSY38AcGrZK/jxPpePkXik=";
+  };
+
+  patches = lib.optional (
+    stdenv.isDarwin && lib.versionOlder stdenv.hostPlatform.darwinMinVersion "11"
+  ) ./dont-use-the-uniformtypeidentifiers-framework.patch;
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+  ] ++ lib.optionals stdenv.cc.isClang [ stdenv.cc.cc.libllvm ];
+
+  buildInputs =
+    [
+      capstone
+      dbus
+      freetype
+      tbb
+    ]
+    ++ lib.optionals withWayland [
+      libglvnd
+      libxkbcommon
+      wayland
+      wayland-protocols
+    ]
+    ++ lib.optionals (stdenv.isDarwin || !withWayland) [ glfw ]
+    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.AppKit ]
+    ++ lib.optionals (stdenv.isDarwin && lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11") [
+      darwin.apple_sdk.frameworks.UniformTypeIdentifiers
+    ];
+
+  cmakeFlags = [
+    "-DDOWNLOAD_CAPSTONE=off"
+    "-DTRACY_STATIC=off"
+  ] ++ lib.optional (stdenv.isLinux && !withWayland) "-DLEGACY=on";
```

Comment:
This flag let me open the native file selector on linux using GTK. Currently it fails.
```nix
++ lib.optional stdenv.isLinux "-DGTK_FILESELECTOR=ON";
```

Reply:
It works fine for me on GNOME (the GTK4 file chooser is opened). Seems like you are missing a [backend](https://wiki.archlinux.org/title/XDG_Desktop_Portal#List_of_backends_and_interfaces) for the XDG Desktop Portal.

Added an option to use the GTK file selector to the Nix derivation. Note that it is `false` by default, as that's what upstream [expects](https://github.com/wolfpld/tracy/blob/v0.11.0/profiler/CMakeLists.txt#L4).